### PR TITLE
Metadata

### PR DESCRIPTION
**Pull request by TomBrouws**
There's now a metadata filter dialog through which genomes are filtered by data. You can select none, some or all options to filter.

### CODE REVIEW

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 1]
```diff
@@ -134,10 +139,15 @@ private void setUpUi() {
 	public void makeGraphFromFolder() {
 		try {
 			File folder = FileSelector.selectFolder("Select data folder", this);
-			File[] files = FileSelector.getFilesFromFolder(folder, ".node.graph", ".edge.graph",".nwk");
-			makeGraph(files[0], files[1], files[2]);
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: *Auto-generated comment*

    Line is longer than 120 characters (found 125).

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 2]
```diff
@@ -134,10 +139,15 @@ private void setUpUi() {
...
-		} catch (FileSelectorException | NullPointerException exception) {
-			if (DialogUtil.confirm("Error!", "Your file was not found. Want to try again?")) {
+
+			File[] files = FileSelector.getFilesFromFolder(folder, ".node.graph", ".edge.graph", ".nwk", ".txt");
+			makeGraph(files[0], files[1], files[2], files[3]);
+		} catch (ArrayIndexOutOfBoundsException exception) {
+			if (DialogUtil.confirm("Error!", "Some necessary files were not found. Want to select a new folder?")) {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: *Auto-generated comment*

    Line is longer than 120 characters (found 128).

File: src/main/java/tudelft/ti2806/pl3/Application.java (hunk 1 above)
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: *Auto-generated comment*

    Line is longer than 120 characters (found 125).

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 3]
```diff
@@ -213,6 +224,21 @@ public void makePhyloTree(File input) {
 	}
 
 	/**
+	 * Load the metadata from a separate file.
+	 */
+	public void loadMetaData() {
+		try {
+			File metaFile = FileSelector.selectFile("Select metadata file", this, ".txt");
+			gd.loadMetaData(metaFile);
+			System.out.println(gd.getGenomes().get(0).getAge());
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: sysout

**Comment by kwendel** [comment]
Agree with Boris. After those fixed its good for merging

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 4]
```diff
@@ -59,6 +60,9 @@
 	private SideBarController sideBarController;
 	private ZoomBarController zoomBarController;
 	private FindgenesController findgenesController;
+	private MetaFilterController metaFilterController;
+
+	private GraphDataRepository gd;
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: never used

File: src/main/java/tudelft/ti2806/pl3/Application.java [hunk 5]
```diff
@@ -213,6 +224,21 @@ public void makePhyloTree(File input) {
 	}
 
 	/**
+	 * Load the metadata from a separate file.
+	 */
+	public void loadMetaData() {
+		try {
+			File metaFile = FileSelector.selectFile("Select metadata file", this, ".txt");
+			gd.loadMetaData(metaFile);
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: why don't you use the graphDataRepository variable?

File: src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        }
+    }
+
+    /**
+     * Opens the dialog which lets the user select a gene.<br>
+     * After a gene is selected, it will do a call to the {@link GraphController} to navigate to the start of that
+     * gene.
+     */
+    public void openDialog() {
+        if (metaFilterView == null) {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java]: why don't you just create it in the constructor?

File: src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            metaFilterView = new MetaFilterView();
+        }
+
+        JOptionPane
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java]: huge method!

File: src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+            return new IntRange(0, 0);
+        }
+    }
+
+    /**
+     * Opens the dialog which lets the user select a gene.<br>
+     * After a gene is selected, it will do a call to the {@link GraphController} to navigate to the start of that
+     * gene.
+     */
+    public void openDialog() {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterController.java]: Huge method. Try breaking it in smaller pieces.

File: src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterView.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+		return isolationDate.getText();
+	}
+
+	private final JTextField strainLocation;
+	private final JTextField isolationDate;
+
+	/**
+	 * Construct the view for the meta filter dialog.
+	 */
+	public MetaFilterView() {
```
Comment (borismattijssen) [inline src/main/java/tudelft/ti2806/pl3/metafilter/MetaFilterView.java]: Large constructor. Use some smaller methods to create different fields.

**Comment by borismattijssen** [comment]
Please look at comments. The rest looks fine.

File: src/main/java/tudelft/ti2806/pl3/Application.java (hunk 1 above)
Comment (TomBrouws) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: Yeah this line doesn't exist anymore :see_no_evil:

File: src/main/java/tudelft/ti2806/pl3/Application.java (hunk 2 above)
Comment (TomBrouws) [inline src/main/java/tudelft/ti2806/pl3/Application.java]: Is 107

**Comment by TomBrouws** [comment]
Thanks for the comments! Everything should be fine now.

**Pull request merged**